frontend/src/metabase/query_builder/components/notebook/lib/steps.ts

```diff
@@ -304,15 +302,36 @@ function getStageSteps(
       }
       steps.splice(i, 1);
     }
-    // revert the previewQuery for this step
-    if (step.revert) {
-      previewQuery = step.revert(
-        previewQuery,
-        step.stageIndex,
-        step.itemIndex ?? undefined,
-      );
-    }
   }
 
   return { steps, actions };
 }
+function getPreviewQuery(
+  query: Lib.Query,
+  stageIndex: number,
+  steps: NotebookStep[],
+  stepIndex: number,
+): Lib.Query {
+  const previewQuery = _.range(
```

Comment:
nit: maybe name both like `queryWithoutNextStages` and `queryWithoutNextSteps` or something like this

Reply:
used `queryWithoutNextStages` and `queryWithoutNextStagesAndSteps`

Reply:
93a896bf640e4921751a9418ba2952bc023aecb5

cleaner, indeed